### ISSUE TITLE
Help create GraphQL query to filter content items by culture

### ISSUE THREAD

**Issue by minhdn2** [OrchardCMS/OrchardCore#6136]
Hi,

I have multiple localized content items with 2 cultures (**id-ID** and **en-US**) in each content items. Now I need to get all the localized versions for **id-ID** of all the content items but I'm not sure how and being blocked by **LocalizationSet**.

`query MyQuery {
  sliderItems(where: {localization: {localizationSet: "", culture: "id-ID"}}) {
    displayText
    localization {
      culture
      localizationSet
    }
  }
}`

 How can I insert multiple values for the required field LocalizationSet? Or how can I ignore this mandatory field?

Please help to check and advise,
Many thanks in advance

**Comment by hishamco** [comment]
> How can I insert multiple values for the required field LocalizationSet?

Why you need to insert multiple values there?!!

@jrestall @carlwoodhouse any idea from GraphQL point of view?

**Comment by carlwoodhouse** [comment]
Possibly need to know a little more of what’s trying to be achieved here ..

Presuming it’s query by multiple localization sets or cultures .. 

I’m not that familiar with the localisation stuff but I assume the filter doesn’t support it out of the box easily ...

But you could always use ‘and’ in the where clause to achieve the he same thing..

Where: { localization: { set: “”, culture: “xx” } and localization { ... }}

That been said (presuming we don’t already I haven’t checked the code) we really should make culture support the in operators

**Comment by minhdn2** [comment]
Basically, I have a pace need to be implemented with localization.

I have been creating content items for this page, and each content items will have 2 localized items, which are **id-ID** and **en-US**.

In default, the page is displayed in **en-US**. So when user choose to display the page in **id-ID**, I will have to get all localized content items in **id-ID** to populate the page.

But I haven't able to do this because the **localizationSet** field is required in GraphQL query.

And since have to get all the **localizationSet** values from all content items, then for-each this list to get all content items in **id-ID** is too complicated, I'm looking for a better solution.

Please help to check and advise,
Many thanks.

**Comment by pbros** [comment]
Hi @minhdn2,

I'm using it like this. Don't know if this helps you or not. I basically get my main element by alias, and then get the localized version using the localizations with the culture filter.
```
query Page {{
    pages: page(where: {{alias: {{alias: ""{alias}""}} }}) {{
        localization {{
            localizations(culture: ""{culture}"") {{
                ... on Page {{
                    title: displayText
                    body: render
                }}
            }}
        }}
    }}
}}
```

**Comment by MikeKry** [comment]
I also need this.. it would be enough to make localizationSet optional. 

UC: 
Query all {language} pages for list views without having to extract localizationSet ids from default culture.

**Comment by sebastienros** [comment]
We have an index for the Localization part, this should be added to graphql to query. Should be two lines of code to add it. (LocalizedContentItemIndex)

**Comment by MikeKry** [comment]
@sebastienros 
I have tried to do it myself, but I can't find where are GraphQL content type filters are defined in Orchard.. (I am pretty new to this project). If you can point me in right direction, I can try to implement and create a PR.

**Comment by agriffard** [comment]
Fixed by #7470
